jetty-websocket/jetty-websocket-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -337,6 +369,82 @@ private void onTextFrame(Frame frame, Callback callback)
         acceptMessage(frame, callback);
     }
 
+    @Override
+    public boolean isDemanding()
+    {
+        return true;
+    }
+
+    public void suspend()
+    {
+        synchronized (this)
+        {
+            switch(state)
+            {
+                case DEMANDING:
+                    state = SuspendState.SUSPENDING;
+                    break;
+
+                case SUSPENDED:
+                case SUSPENDING:
+                    throw new IllegalStateException("Already Suspended");
+
+                default:
+                    throw new IllegalStateException();
+            }
+        }
+    }
+
+    public void resume()
+    {
+        Callback onFrame;
+        synchronized (this)
+        {
+            onFrame = suspendedCallback;
```

Comment:
This should probably only be done in suspended state and all other states a non null is ISE